visca_over_ip/camera.py

```diff
@@ -0,0 +1,290 @@
+import socket
+from typing import Optional
+
+from visca_over_ip.exceptions import ViscaException, NoQueryResponse
+
+
+SEQUENCE_NUM_MAX = 2 ** 32 - 1
+
+
+class Camera:
+    """
+    Represents a camera that has a VISCA-over-IP interface.
+    Provides methods to control a camera over that interface.
+
+    Only one camera can be connected on a given port at a time.
+    If you wish to use multiple cameras, you will need to switch between them (use :meth:`close_connection`)
+    or set them up to use different ports.
+    """
+    def __init__(self, ip, port=52381):
+        """:param ip: the IP address or hostname of the camera you want to talk to.
+        :param port: the port number to use.
+        """
+        self._location = (ip, port)
+        self._sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)  # for UDP stuff
+        self._sock.bind(('', port))
+        self._sock.settimeout(0.1)
+
+        self.num_missed_responses = 0
+        self.sequence_number = 0  # This number is encoded in each message and incremented after sending each message
+        self.num_retries = 5
+        self.reset_sequence_number()
+        self._send_command('00 01')  # clear the camera's interface socket
+
+    def _send_command(self, command_hex: str, query=False) -> Optional[bytes]:
```

Comment:
This method is kind of... a lot. Sorry. It works pretty reliably though, and it really simplifies the code in all the public methods (the ones that don't start with "_")